src/pudl/transform/classes.py

```diff
@@ -252,9 +252,54 @@ def normalize_strings(col: pd.Series, params: StringNormalization) -> pd.Series:
 normalize_strings_multicol = multicol_transform_factory(normalize_strings)
 
 
+################################################################################
+# Enforce Snake Case
+################################################################################
+
+
+class EnforceSnakeCase(TransformParams):
+    """Boolean parameter for :func:`enforce_snake_case`."""
+
+    enforce_snake_case: bool
+
+
+def enforce_snake_case(
+    col: pd.Series, params: EnforceSnakeCase = EnforceSnakeCase(enforce_snake_case=True)
```

Comment:
It's generally considered bad form to do these instantiations in the function call signature itself, since it means that `EnforceSnakeCase()` will get called on import, rather than only when the program is run. So if there's a problem with that class / initialization, the whole package becomes unimportable, which is unpleasant for debugging. A cleaner pattern with complex default values is to allow `None` and then check if `None` as the first line in the function, and instantiate the default value there if necessary.